cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/query/impl/CassandraParameterMetadataBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.query.impl;
+
+import java.util.Map;
+
+import com.datastax.driver.core.ColumnDefinitions;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+
+import org.hibernate.boot.model.naming.Identifier;
+import org.hibernate.engine.query.spi.OrdinalParameterDescriptor;
+import org.hibernate.engine.query.spi.ParameterMetadata;
+import org.hibernate.mapping.Column;
+import org.hibernate.mapping.Table;
+import org.hibernate.ogm.dialect.query.spi.ParameterMetadataBuilder;
+import org.hibernate.type.Type;
+
+/**
+ * {@link ParameterMetadataBuilder} for native Cassandra CQL queries.
+ * This implementation delegates parsing to the server cluster, so no string positional information.
```

Comment:
confusing two things here maybe?. string positional info means 'character offset withing string', used for parse error reporting, which isn't available because the parse is server side rather than client side. That's different to 'ordinal position of parameter within array' for bind purposes, which is a whole other problem.